inifile.cc

```diff
@@ -132,35 +128,31 @@ inifile_load_file(gbfile* fin, inifile_t* inifile, const char* myname)
               qPrintable(gbinipathname));
       }
 
-      sec = (inifile_section_t*) xcalloc(1, sizeof(*sec));
-
-      sec->name = xstrdup(cin);
-      QUEUE_INIT(&sec->entries);
-      ENQUEUE_TAIL(&inifile->secs, &sec->Q);
-      inifile->isecs++;
+      inifile->secs.append(inifile_section_t());
+      sec = &inifile->secs.last();
+      sec->name.reset(xstrdup(cin), xfree);
     } else {
       if (sec == nullptr) {
         fatal("%s: missing section header in '%s'.\n", myname,
               qPrintable(gbinipathname));
       }
 
-      inifile_entry_t* entry = (inifile_entry_t*) xcalloc(1, sizeof(*entry));
-      ENQUEUE_TAIL(&sec->entries, &entry->Q);
-      sec->ientries++;
+      sec->entries.append(inifile_entry_t());
+      inifile_entry_t* entry = &sec->entries.last();
 
       char* cx = strchr(cin, '=');
       if (cx != nullptr) {
         *cx = '\0';
         cin = lrtrim(cin);
       }
 
-      entry->key = xstrdup(cin);
+      entry->key.reset(xstrdup(cin), xfree);
 
       if (cx != nullptr) {
         cx = lrtrim(++cx);
-        entry->val = xstrdup(cx);
+        entry->val.reset(xstrdup(cx), xfree);
       } else {
-        entry->val = xstrdup("");
+        entry->val.reset(xstrdup(""), xfree);
```

Comment:
That goes to a gross length to make entry->val = QString() - and whatever code looking at this that's prepared to make the distinction of an empty string and a null string pointer probably becamse accidentally smarter when we moved to QString anyway.